pkg/connector/role.go

```diff
@@ -308,9 +318,27 @@ func (r *roleBuilder) cacheRoleAssignments(ctx context.Context, subscriptionID s
 			return err
 		}
 
-		assignments, _ := r.cacheGet(subscriptionID)
-		r.cacheSet(subscriptionID, append(assignments, page.Value...))
+		assignments, _ := r.cache.GetOrSet(subscriptionID, func() (map[string]*armauthorization.RoleAssignment, error) {
+			return make(map[string]*armauthorization.RoleAssignment), nil
+		})
+
+		for _, assignment := range page.Value {
+			if assignment.Properties == nil && assignment.Properties.RoleDefinitionID != nil {
+				l.Warn("baton-azure-infrastructure: role assignment properties are nil")
+				continue
+			}
+
+			assignments[*assignment.Properties.RoleDefinitionID] = assignment
```

Comment:
won't this overwrite the value if there are multiple assignments in the page for the same role?

Reply:
That's right, I will fix it, I created to skip unused roles, but forgot about the Grants Method